src/api/object.js

```diff
@@ -1,33 +1,259 @@
+      }
+      if (!options) {
+        options = {}
+      }
+
+      let tmpObj = {
+        Data: null,
+        Links: []
+      }
+
+      if (Buffer.isBuffer(obj)) {
+        if (!options.enc) {
+          tmpObj = { Data: obj.toString(), Links: [] }
+        }
+      } else if (obj.multihash) {
+        tmpObj = {
+          Data: obj.data.toString(),
+          Links: obj.links.map((l) => { return l.toJSON() })
+        }
+      } else if (typeof obj === 'object') {
+        tmpObj.Data = obj.Data.toString()
+      } else {
+        return callback(new Error('obj not recognized'))
+      }
+
+      let buf
+      if (Buffer.isBuffer(obj) && options.enc) {
+        buf = obj
+      } else {
+        buf = new Buffer(JSON.stringify(tmpObj))
+      }
+      const enc = options.enc || 'json'
+
+      send('object/put', enc, null, buf, (err, result) => {
+        if (err) {
+          return callback(err)
+        }
+
+        if (Buffer.isBuffer(obj)) {
+          if (!options.enc) {
+            obj = { Data: obj, Links: [] }
+          } else {
+            obj = JSON.parse(obj.toString())
+          }
+        }
+        let node
+        if (obj.multihash) {
+          node = obj
+        } else {
+          node = new DAGNode(obj.Data, obj.Links)
+        }
+
+        if (node.toJSON().Hash !== result.Hash) {
+          return callback(new Error('Stored object was different from constructed object'))
+        }
+
+        callback(null, node)
+      })
+    }),
+    data: promisify((multihash, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+        options = {}
+      }
+      if (!options) {
+        options = {}
+      }
+      multihash = cleanMultihash(multihash, options)
+
+      send('object/data', multihash, null, null, (err, result) => {
+        if (err) {
+          return callback(err)
+        }
+
+        result.pipe(bl(callback))
+      })
+    }),
+    links: promisify((multihash, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+        options = {}
+      }
+      if (!options) {
+        options = {}
+      }
+      multihash = cleanMultihash(multihash, options)
+
+      send('object/links', multihash, null, null, (err, result) => {
+        if (err) {
+          return callback(err)
+        }
+
+        let links = []
+
+        if (result.Links) {
+          links = result.Links.map((l) => {
+            return new DAGLink(l.Name, l.Size, new Buffer(bs58.decode(l.Hash)))
+          })
+        }
+        callback(null, links)
+      })
+    }),
+    stat: promisify((multihash, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+        options = {}
+      }
+      if (!options) {
+        options = {}
+      }
+      multihash = cleanMultihash(multihash, options)
+
+      send('object/stat', multihash, null, null, callback)
+    }),
+    new: promisify((callback) => {
```

Comment:
I tried all the other layouts through the CLI and only `unixfs-dir` works. It seems one of those deprecated features that no one ever used to even notice. I updated the spec accordingly.